docs/enhancements/arm64-cpu-architecture.md

```diff
@@ -0,0 +1,149 @@
+
+```
+cpu_architecture:
+   type: string
+   default: 'x86_64'
+   description: The CPU architecture of the image (x86_64/arm64/etc)
+```
+
+When generating a discovery ISO, the proper RHCOS/Release images should be used according to CPU architecture specified in the cluster. The `cpu_architecture` property should be stored in `infraenv` after the ISO is generated.
+
+### CPU_ARCHITECTURES environment variable
+
+The supported CPU architectures will be defined in CPU_ARCHITECTURES environment variable. The default should be: 'x86_64,arm64'.
+As arm64 is not supported for older OCP versions (<4.9), it could be validated against OPENSHIFT_VERSIONS only for OCP >=4.9.
+
+### User Stories
+
+#### Story 1
+
+As an Assisted-Installer user, I need to install a cluster on non x86_64 hosts.
+
+#### Story 2
+
+As an Assisted-Installer user, I need to install a Single-Node OpenShift cluster on an arm64 host.
+
+### Implementation Details/Notes/Constraints [optional]
+
+* The current focus is for SNO flows, but other scenarios could be supported.
+
+* Late binding is not covered by this enhancement, but decoupling the discovery ISO generation from the cluster should be simple enough. I.e. adding `cpu_architecture` property to `infraenv_create_params` and override the architecture defined in the cluster. Also, architecture validation should be probably added to host discovery conditions (i.e. to ensure that the host CPU architecture is similar to the cluster's architecture).
+
+### Open Questions
+
+#### kube-api considerations
+
+##### ClusterImageSet
+Since we extract the openshift-baremetal-install binary in the assisted-service (to create the manifests of the cluster), the x86_64 OCP release image has to be used. Which means that both arm64 and x86_64 release images must be provided.
+  * Should we perhaps add another URL for the x86_64 release image in ClusterImageSet?
+  * Maybe we could assume that a x86_64 release image is available by default for each 'y-stream' version?
+  * As a workaround, we could use something like [qemu-user-static](https://github.com/multiarch/qemu-user-static), which enables invocation of multi-architecture containers using QEMU. I.e it might be sufficient just for running the arm64 openshift-baremetal-install binary, though not sure we want to rely on such solution.
+
+##### AgentServiceConfig
+`osImages` property currently contains a single set of images for each OCP version.
+E.g.
+```
+osImages:
+- openshiftVersion: '4.8'
+  rootFSUrl: https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.8/4.8.2/rhcos-live-rootfs.x86_64.img
+  url: https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/4.8/4.8.2/rhcos-4.8.2-x86_64-live.x86_64.iso
+  version: 48.84.202107202156-0
+```
+
+We should consider API changes to support multiple architectures for each version. Perhaps just add `cpuArchitecture` property for each image, i.e. multiple images for each version. Would need to modify parsing handling obviously.
```

Comment:
I like the idea of just adding a `cpuArchitecture` field but I also think it would be better for us to be consistent with the structure used elsewhere.

That said, the current structure is already inconsistent with the `OPENSHIFT_VERSIONS` one so I guess adding the field should do it